### PR TITLE
6774 6780 downloads and cache discovery details viewer

### PR DESCRIPTION
**Pull request by wschaeferB**

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java [hunk 1]
```diff
@@ -284,17 +284,50 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
...
+            return Lookups.fixed(artifact, content);
+        }
+
+    }
+
+    /**
+     * Private helper method to allow content specified in a path id attribute
+     * to be retrieved.
+     *
+     * @param artifact The artifact which content may be specified as a tsk path
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java:311]: "which content" => "for which content"

File: Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java [hunk 2]
```diff
@@ -284,17 +284,50 @@ public BlackboardArtifactNode(BlackboardArtifact artifact) {
...
+     */
+    private static Content getPathIdFile(BlackboardArtifact artifact) throws ExecutionException {
+        try {
+            for (BlackboardAttribute attribute : artifact.getAttributes()) {
+                if (attribute.getAttributeType().getTypeID() == BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PATH_ID.getTypeID()) {
+                    return contentCache.get(attribute.getValueLong(), () -> artifact.getSleuthkitCase().getContentById(attribute.getValueLong()));
+                }
+            }
+        } catch (TskCoreException ex) {
+            logger.log(Level.WARNING, MessageFormat.format("Error getting content for path id attrbiute for artifact: ", artifact.getId()), ex); //NON-NLS
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java:328]: I think that we can simplify this code slightly by taking advantage of this BlackboardArtifact method:
BlackboardAttribute getAttribute(BlackboardAttribute.Type attributeType).

**Review by rcordovano — CHANGES_REQUESTED** [review]

File: Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java (hunk 1 above)
Comment (wschaeferB) [reply Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java]: Thank you.

File: Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java (hunk 2 above)
Comment (wschaeferB) [reply Core/src/org/sleuthkit/autopsy/datamodel/BlackboardArtifactNode.java:327]: Good catch. Thank you.

**Review by rcordovano — APPROVED** [review]

**Pull request merged**